cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizer.java

```diff
@@ -238,6 +245,20 @@ private Boolean doEnforce(AuthorizationPrivilege authorizationPrivilege) {
                       Collections.singleton(authorizationPrivilege.getAction()));
   }
 
+  private void invalidateCache(EntityId entityId, Principal principal, Set<Action> actions) {
```

Comment:
The cache needs to be invalidated when an entity is deleted too else in scenario where entity is deleted and created again with same name it will have the privilege for users in cache.